provider/maas/environ.go

```diff
@@ -899,9 +899,20 @@ func (*maasEnviron) MaintainInstance(args environs.StartInstanceParams) error {
 }
 
 // StartInstance is specified in the InstanceBroker interface.
-func (environ *maasEnviron) StartInstance(args environs.StartInstanceParams) (
-	*environs.StartInstanceResult, error,
-) {
+func (environ *maasEnviron) StartInstance(
+	args environs.StartInstanceParams,
+) (_ *environs.StartInstanceResult, err error) {
+
+	zoneIndependent := true
+	defer func() {
+		if err != nil && zoneIndependent {
+			err = &common.StartInstanceError{
+				Err:             err,
+				ZoneIndependent: true,
+			}
+		}
```

Comment:
Why not just always wrap but pass in:
ZoneIndependent: zoneIndependent
?

Reply:
Just felt like unnecessary layering, but it's irrelevant now anyway, as I've dropped the defer.